agenta-backend/agenta_backend/services/db_manager.py

```diff
@@ -607,26 +635,48 @@ async def get_user_object(user_uid: str) -> UserDB:
 
     user = await engine.find_one(UserDB, UserDB.uid == user_uid)
     if user is None:
-        org = OrganizationDB()
-        return UserDB(uid="0", organization_id=org)
-    return user
+        create_user = UserDB(uid="0")
+        await engine.save(create_user)
 
+        org = OrganizationDB(owner=create_user)
+        await engine.save(org)
 
-async def get_user_organization(user_id: str) -> OrganizationDB:
-    """Get the user organization object from the database.
+        create_user.organizations.append(org.id)
+        await engine.save(create_user)
+        await engine.save(org)
+
+        return create_user
+    else:
+        return user
+
+
+async def get_user_organizations(
+    user_id_or_user: Union[str, UserDB] = None
+) -> List[OrganizationDB]:
+    """Get the list of the user's organizations from the database.
 
     Arguments:
-        user_id (str): The user unique identifier
+        user_id_or_user (Union[str, UserDB]): Either the user's unique identifier (str) or the user object (UserDB).
 
     Returns:
-        OrganizationDB: instance of user organization
+        List[OrganizationDB]: list of user's organizations.
     """
+    print("This function was called.")
 
-    user = await get_user_object(user_id)
-    organization = await engine.find_one(
-        OrganizationDB, OrganizationDB.id == user.organization_id
-    )
-    return organization
+    if isinstance(user_id_or_user, str):
```

Comment:
why do we have this? why are we not sure about the input type in this case?

Reply:
@devgenix 

Initially, the function uses the user_id, I modified it to accept both so that in cases where we already have fetched user and need to get their organization we can just pass the organization instead of passing the id and query for the user again in the get_user_organizations function.